pybop/parameters/priors.py

```diff
@@ -90,6 +122,98 @@ def rvs(self, size=1, random_state=None):
             loc=self.loc, scale=self.scale, size=size, random_state=random_state
         )
 
+    def __call__(self, x):
+        """
+        Evaluates the distribution at x.
+
+        Parameters
+        ----------
+        x : float
+            The point(s) at which to evaluate the distribution.
+
+        Returns
+        -------
+        float
+            The value(s) of the distribution at x.
+        """
+        return self.evaluate(x)
+
+    def evaluate(self, x):
+        """
+        Evaluates the distribution at x.
+
+        Parameters
+        ----------
+        x : float
+            The point(s) at which to evaluate the distribution.
+
+        Returns
+        -------
+        float
+            The value(s) of the distribution at x.
+        """
+        inputs = self.verify(x)
+        return self._evaluate(inputs)
+
+    def _evaluate(self, x):
+        """
+        Evaluates the distribution at x.
+
+        Parameters
+        ----------
+        x : float
+            The point(s) at which to evaluate the distribution.
+
+        Returns
+        -------
+        float
+            The value(s) of the distribution at x.
+        """
+        return self.logpdf(x)
+
+    def evaluateS1(self, x):
+        """
+        Evaluates the first derivative of the distribution at x.
+
+        Parameters
+        ----------
+        x : float
+            The point(s) at which to evaluate the first derivative.
+
+        Returns
+        -------
+        float
+            The value(s) of the first derivative at x.
+        """
+        inputs = self.verify(x)
+        return self._evaluateS1(inputs)
+
+    def _evaluateS1(self, x):
+        """
+        Evaluates the first derivative of the distribution at x.
+
+        Parameters
+        ----------
+        x : float
+            The point(s) at which to evaluate the first derivative.
+
+        Returns
+        -------
+        float
+            The value(s) of the first derivative at x.
+        """
+        raise NotImplementedError
+
+    def verify(self, x):
```

Comment:
I would not expect a dictionary to be passed here, so let's remove the verify function unless there is a particular reason for it?

Reply:
Given the call stack to this point, it's likely to have this as a dict, as `parameters.verify(x)` within `cost.evaluate` provides a dict for the `cost.compute()`. In this situation, the `x` dict for the prior calculation in the LogPosterior class needs to be converted from dict to ndarray for the prior methods.

I think this is probably the best spot for the conversion, as the alternative would be to individually apply the conversion to each `cost.compute`, or refactor the prior methods to accept dicts. Do you have a preference?

Reply:
The dictionary if statement is not currently checked by tests. If there is an easy test for this, it would be good to add. Otherwise, that might be a good argument towards removing the verify function (as in, if there is no easy test for that, it is unlikely that a user will inadvertently pass a dictionary).